Fix table C3

hannahsfraser opened the pull request:
As per issue #26 I have added sorensen and random effects results to this table - note that I haven't been able to check this because I cant seem to run any many analysts stuff on my computer anymore.

hannahsfraser committed:
Update SM3_ExplainingDeviation.qmd

As per issue #26 I have added sorensen and random effects results to this table - note that I haven't been able to check this because I cant seem to run any many analysts stuff on my computer anymore.

egouldo commented on supp_mat/SM3_ExplainingDeviation.qmd:
```diff
@@ -981,13 +981,13 @@ patchwork::wrap_plots(yi_cont_plots,heights = 4, byrow = TRUE) +
                               TRUE ~ dataset),
           model_name = forcats::as_factor(model_name) %>% 
-            forcats::fct_relevel(c("box_cox_rating_cat", 
+            forcats::fct_relevel(c("box_cox_rating_cat_no_int", 
```
good pickup

egouldo replied:
Updated to reflect fixes in [aee7c5d](https://github.com/egouldo/ManyAnalysts/pull/66/commits/aee7c5dcc17f09d2366b290bcc5d8db4503a5553)

egouldo commented on supp_mat/SM3_ExplainingDeviation.qmd:
```diff
@@ -981,13 +981,13 @@ patchwork::wrap_plots(yi_cont_plots,heights = 4, byrow = TRUE) +
   filter(exclusion_set == "complete", 
          dataset == "blue tit",
-         model_name %nin% c("MA_mod", "box_cox_rating_cat_no_int")) %>% 
+         model_name %nin% c("box_cox_rating_cont", "box_cox_rating_cat_no_int","sorensen_glm" )) %>% 
```
I think this isn't going to work because %nin% is *NOT IN* so this will remove the sorensen's

egouldo replied:
fixed in [aee7c5d](https://github.com/egouldo/ManyAnalysts/pull/66/commits/aee7c5dcc17f09d2366b290bcc5d8db4503a5553)

egouldo commented on supp_mat/SM3_ExplainingDeviation.qmd:
```diff
@@ -981,13 +981,13 @@ patchwork::wrap_plots(yi_cont_plots,heights = 4, byrow = TRUE) +
+         model_name %nin% c("box_cox_rating_cont", "box_cox_rating_cat_no_int","sorensen_glm" )) %>% 
   bind_rows({euc_yi_results %>% 
-      filter(model_name %nin% c("MA_mod", "box_cox_rating_cat_no_int"))}) %>% 
+      filter(model_name %nin% c("box_cox_rating_cont", "box_cox_rating_cat_no_int","sorensen_glm", "uni_mixed_effects"))}) %>% 
```
Same as previous comment, `%nin%` removes anything in `c()` from the data frame.

egouldo replied:
Fixed in [aee7c5d](https://github.com/egouldo/ManyAnalysts/pull/66/commits/aee7c5dcc17f09d2366b290bcc5d8db4503a5553)

egouldo committed:
#64 revert filtering of models, and add MA_mod_mv to list to filter out

egouldo commented on supp_mat/SM3_ExplainingDeviation.qmd:
```diff
@@ -981,20 +981,26 @@ patchwork::wrap_plots(yi_cont_plots,heights = 4, byrow = TRUE) +
+              `Deviation explained by Sorensen's index` =  "sorensen_glm",
+              `Deviation explained by inclusion of random effects` =  "uni_mixed_effects")
+  ) %>% 
   semi_join(
     {bind_rows(yi_singularity_convergence_sorensen_mixed_mod, 
                yi_convergence_singularity) %>% 
```
This is the reason why Sorensen's is being removed, because it seems that none of the Sorensen's models have converged/are non-singular/can have the SD of the random effects calculated. Note that there is no column `SD_calc` for the models in yi_singularity_convergence_sorensen_mixed_mod, so we end up with `NA`.

egouldo replied:
resolved in [ef78952](https://github.com/egouldo/ManyAnalysts/pull/66/commits/ef78952f98f4de45a70e597a9af3438a3678e9ef)

egouldo committed:
#64 permit SD_calc == NA generated on binding  with
- code erroneously filtered out Sorensen's / mixed effects univarirate models which didn't have their SD_calc check performed

egouldo committed:
Merge branch '42-mv-fns-2-pkg' into hannahsfraser-patch-3

egouldo commented:
> As per issue #26 I have added sorensen and random effects results to this table - note that I haven't been able to check this because I cant seem to run any many analysts stuff on my computer anymore.

I've fixed this now. Heads up, you need to install the latest version of the package on main:

`devtools::install_github("egouldo/ManyEcoEvo")`
